src/back/TicketManager/src/PaymentService/TicketManager.PaymentService.Domain/Payments/Payment.cs

```diff
@@ -7,15 +7,47 @@ public class Payment: IAggregateRoot<Guid>, IOptimisticConcurrent
     public Guid Id { get; private init; }
     public PaymentStatus PaymentStatus { get; private set; }
     public DateTime DateCreated { get; private init; }
+    public DateTime DateToExpire { get; private init; }
+    private const int ExpirationTimeInMinutes = 30;
     public DateTime DateModified { get; set; }
 
     public Payment()
     {
         Id = Guid.NewGuid();
         PaymentStatus = PaymentStatus.Created;
         DateCreated = DateTime.UtcNow;
+        DateToExpire = DateCreated.AddMinutes(ExpirationTimeInMinutes);
     }
 
+    public virtual bool ConfirmPayment()
+    {
+        if (PaymentAlreadyDecidedOrExpired())
+        {
+            return false;
```

Comment:
Throw exception here instead of returning bool.

Reply:
Addressed in c7c6740e.